spark2/src/main/java/org/apache/iceberg/spark/source/Reader.java

```diff
@@ -338,12 +330,50 @@ public boolean enableBatchRead() {
 
       boolean hasNoDeleteFiles = tasks().stream().noneMatch(TableScanUtil::hasDeletes);
 
+      boolean batchReadsEnabled = batchReadsEnabled(allParquetFileScanTasks, allOrcFileScanTasks);
+
       this.readUsingBatch = batchReadsEnabled && hasNoDeleteFiles && (allOrcFileScanTasks ||
           (allParquetFileScanTasks && atLeastOneColumn && onlyPrimitives));
     }
     return readUsingBatch;
   }
 
+  private boolean batchReadsEnabled(boolean isParquetOnly, boolean isOrcOnly) {
+    if (isParquetOnly) {
+      return isVectorizationEnabled(FileFormat.PARQUET);
+    } else if (isOrcOnly) {
+      return isVectorizationEnabled(FileFormat.ORC);
+    } else {
+      return false;
+    }
+  }
+
+  public boolean isVectorizationEnabled(FileFormat fileFormat) {
```

Comment:
Sounds like we would benefit from a class that takes read options, session config and table properties and returns the value we should use. Maybe, worth adding in the future.

Reply:
Agreed, it may also be good to factor out Iceberg session confs into a class of its own along with the defaults. We have three right now.
```
spark.sql.iceberg.vectorization.enabled
spark.sql.iceberg.check-ordering
spark.sql.iceberg.check-nullability
```

Also probably worth adding these configs to the documentation

Reply:
Yep, for sure.